pandas/core/generic.py

```diff
@@ -488,16 +488,16 @@ def _get_index_resolvers(self):
             d.update(self._get_axis_resolvers(axis_name))
         return d
 
-    def _get_space_character_free_column_resolvers(self):
-        """Return the space character free column resolvers of a dataframe.
+    def _get_special_character_free_column_resolvers(self):
```

Comment:
Slightly orthogonal to this PR but this function only works for DataFrames right? If so might make sense to move from generic to DataFrame (unless it can actually operate on a Series)

Reply:
Technically yes, it can operate on Series. However, since eval/query are only methods available for DataFrames, it will never be used on a Series by a library function unless someones explicitly calls it. 

So, shall I move it?

Reply:
Keep here for now can move in a follow up